test/integration/integration_admin_test.cc

```diff
@@ -122,6 +124,20 @@ TEST_F(IntegrationTest, Admin) {
                                                 Http::CodecClient::Type::HTTP1);
   EXPECT_TRUE(response->complete());
   EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+
+  response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/listeners", "",
+                                                Http::CodecClient::Type::HTTP1);
+  std::list<std::string> listener_addresses;
+  int listener_index = 0;
+  while (test_server_->server().getListenSocketByIndex(listener_index) != nullptr) {
+    listener_addresses.push_back(
+        test_server_->server().getListenSocketByIndex(listener_index)->localAddress()->asString());
+    ++listener_index;
+  }
+  EXPECT_TRUE(response->complete());
+  EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+  EXPECT_STREQ(Json::Factory::listAsJsonString(listener_addresses).c_str(),
```

Comment:
Rather than comparing the returned string, you could parse the response via RapidJSON and afterwards compare with what `getListenSocketByIndex` says. It's a bit more declarative, since you're not relying on the serialization in use.